src/functions.R

```diff
@@ -227,3 +268,231 @@ fetch_wqp_data <- function(site_counts_grouped, char_names, wqp_args = NULL,
+                                 name = paste0(target_string,
+                                               "_",
+                                               gsub(pattern = "-",
+                                                    replacement = "",
+                                                    x = Sys.Date()),
+                                               ".rds"))
+    
+    # Make the Google Drive link shareable: anyone can view
+    drive_share(out_file_stable, role = "reader", type = "anyone")
+  }
+  
+  # Now remove the local file after upload is complete
+  file.remove(file_local_path)
+  
+}
+
+
+#' @title Retrieve a dataset from Google Drive
+#' 
+#' @description
+#' A function to retrieve a dataset from Google Drive after it has been uploaded
+#' in a previous step.
+#' 
+#' @param target The name of the target to be exported (as an object not a string).
+#' 
+#' @param local_folder A string specifying the folder where the file should be
+#' downloaded.
+#' 
+#' @param drive_folder A string specifying the folder location on Google Drive
+#' containing the file to be downloaded.
+#' 
+#' @param stable Logical value. If TRUE, look for file in the "stable" subfolder
+#' in Google Drive. If FALSE, use the path as provided by the user.
+#' 
+#' @param google_email A string containing the gmail address to use for
+#' Google Drive authentication.
+#' 
+#' @param file_type A string giving the file extension to be used.
+#' 
+#' @param stable_ids  A target containing a tibble with the cols `name` and `id`
+#' to use for retrieving uploaded dataset from Google Drive. ONLY used if stable
+#' is TRUE.
+#' 
+#' @returns 
+#' The dataset after being downloaded and read into the pipeline from Google Drive.
+#' 
+retrieve_data <- function(target, local_folder, drive_folder, stable, 
+                          google_email, file_type = ".rds", stable_ids){
+  
+  # Authorize using the google email provided
+  drive_auth(google_email)
+  
+  # Get target name as a string
+  target_string <- deparse(substitute(target))
+  
+  # Local file download location
+  local_path <- file.path(local_folder, paste0(target_string, ".rds"))
+  
+  # Get file contents of the Google Drive folder specified. If stable == TRUE
+  # then append "stable/" to go to the subfolder for stable products.
+  if(stable){
+    
+    folder_contents <- stable_ids
+    
+  } else{
+    
+    folder_contents <- drive_ls(path = drive_folder)
+    
+  }
+  
+  # Filter the contents to the file requested and obtain its ID
+  drive_file_id <- folder_contents %>%
+    filter(grepl(x = name, pattern = target_string)) %>%
+    pull(id) %>%
+    as_id(.)
+  
+  # Run the download
+  drive_download(file = drive_file_id,
+                 path = local_path,
+                 overwrite = TRUE)
+  
+  # Read dataset into pipeline
+  read_rds(local_path)
+  
+}
+
+
+#' @title Function to retrieve Google Drive IDs for a specified folder.
```

Comment:
Heck yeah! My initial read of this function is that it's overly complicated in a way that it doesn't need to be. Here's some context as to why: this function will be used to populate a .csv that's tracked even though it's not currently (`data/out/chl_stable_drive_ids.csv`) so that downstream users (or us in a different repo) can easily download from Drive without running the entire workflow. If you set a specific version date (either as a config setting, or as a general argument in the function), this function becomes much more straightforward and you can just reduce the dribble by the pattern of the version setting.